### DISCUSSION TITLE
Design of U-Net

### DISCUSSION THREAD

**Discussion by dongyang0122** [Project-MONAI/MONAI#4165] [Q&A]
Hi, I am wondering how the ADN (act--dropout-norm) is designed in the current U-Net implementation as follows.
https://docs.monai.io/en/stable/_modules/monai/networks/nets/unet.html

I would like to add dropout layers in another network. What is the best practice to place dropout layers between other layers? Thanks!

@Nic-Ma @ericspod

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help share some guidance about this question?

Thanks in advance.

**Comment by wyli** [comment]
It might be useful to include the adn_ordering option from convolution https://github.com/Project-MONAI/MONAI/blob/8f5c9924df73f6f3dbb222d6a1263d71b7abd0cd/monai/networks/blocks/convolutions.py#L110
in the unet, such as: https://github.com/Project-MONAI/MONAI/blob/8f5c9924df73f6f3dbb222d6a1263d71b7abd0cd/monai/networks/nets/unet.py#L240-L243

this will bring a bit more flexibilities for activation/dropout/normalisation... cc @ericspod

**Reply by ericspod** [reply]
We can enhance the class to add this argument, that would solve changing the order. @dongyang0122 what do you mean by "add dropout layers in another network"?

**Reply by dongyang0122** [reply]
@ericspod I would like to add dropout layers into another network architecture. not sure where to add those layers (before or after activation layers). I am wondering if you have any guideline or reference about the design order of layers in the U-Net implementation. Thanks!

**Reply by ebrahimebrahim** [reply]
@dongyang0122 
throwing out some links that might be helpful:
- [NDA is the default ordering used in convolution blocks](https://github.com/Project-MONAI/MONAI/blob/05b9d3110ca68d6550e7ecf4553b4adeae997305/monai/networks/blocks/convolutions.py#L263), which are the ones used in the UNet
- [NDA is also the default ordering chosen by the ADN block](https://github.com/Project-MONAI/MONAI/blob/05b9d3110ca68d6550e7ecf4553b4adeae997305/monai/networks/blocks/acti_norm.py#L71)
- [the documentation for the ADN block](https://docs.monai.io/en/stable/networks.html#adn) may be useful

**Reply by ericspod** [reply]
UNet was also [recently updated ](https://github.com/Project-MONAI/MONAI/pull/4185) by @wyli to implement his proposed addition.

I would also recommend the default NDA ordering.

**Comment by dongyang0122** [comment]
@ebrahimebrahim thank you for sharing the links! It would be better if there are any existing experiments or numbers supporting the design.
